code/modules/pda/PDA.dm

```diff
@@ -60,6 +60,8 @@ GLOBAL_LIST_EMPTY(PDAs)
 	var/ownrank = null // this one is rank, never alt title
 
 	var/obj/item/paicard/pai = null	// A slot for a personal AI device
+	// The slot where you can store a pen
+	var/obj/item/held_pen
```

Comment:
You need to account for GC issues here, if held_pen gets deleted you need to clear the reference

Reply:
I feel like I'm missing something. I thought I already accounted for it when removing the pen or when the pda gets destroyed.

Reply:
if the pen gets QDEL'd the PDA won't know that it doesn't exist anymore and will still hold a reference to it.

You could check to see how books handle items stored inside them and ensures GC